config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml

```diff
@@ -17,6 +17,7 @@ spec:
     - otelcol
     - otelcols
     singular: opentelemetrycollector
+  preserveUnknownFields: false
```

Comment:
Is there a reason unknown fields should be allowed?

Reply:
No idea. To be honest, I haven't given it much thought, as it's required to be `false` for the validation webhook to work. Creating a webhook configuration for a CRD that has this set to `true` will just fail.